wagtailio/navigation/models.py

```diff
@@ -0,0 +1,126 @@
+from django.db import models
+
+from modelcluster.fields import ParentalKey
+from modelcluster.models import ClusterableModel
+from wagtail.admin.edit_handlers import (
+    FieldPanel,
+    InlinePanel,
+    MultiFieldPanel,
+    StreamFieldPanel,
+)
+from wagtail.contrib.settings.models import BaseSetting, register_setting
+from wagtail.core.fields import StreamField
+from wagtail.core.models import Orderable
+from wagtail.snippets.edit_handlers import SnippetChooserPanel
+from wagtail.snippets.models import register_snippet
+
+from wagtailio.navigation.blocks import NavItemBlock, NavItemCTA, NavStreamField
+from wagtailio.navigation.forms import MainMenuSnippetForm
+
+
+@register_snippet
+class FooterMenuSnippet(models.Model):
+    name = models.CharField(max_length=255)
+    sections = StreamField(
+        NavStreamField(),
+    )
+
+    panels = [
+        FieldPanel("name"),
+        StreamFieldPanel("sections"),
+    ]
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        verbose_name = "Footer menu"
+
+
+@register_snippet
+class MainMenuItemSnippet(models.Model):
```

Comment:
nitpick: let's call it just `MainMenuItem`, ditto for `MainMenuSnippet` and `FooterMenuSnippet`

Reply:
as an aside.. perhaps `MainMenuSection` is more appropriate, but this is not a blocker

Reply:
Thanks, done some bit of refactoring in 080d08f